include/zephyr/drivers/fuel_gauge.h

```diff
@@ -235,6 +273,35 @@ static inline int z_impl_fuel_gauge_get_prop(const struct device *dev,
 	return api->get_property(dev, props, props_len);
 }
 
+/**
+ * @brief Fetch a battery fuel-gauge block property
+ *
+ * @param dev Pointer to the battery fuel-gauge device
+ * @param props pointer to single fuel_gauge_get_block_property struct where the property struct
+ * field is set by the caller to determine what property is read from the
+ * fuel gauge device into the dst field.
+ * @param dst_len the length of the destination array in bytes
+ * @param dst byte array that will hold the block data that is read from the fuel gauge
+ *
+ * @return return=0 if successful, return < 0 if getting property failed, return 0 on success
+ */
+__syscall int fuel_gauge_get_block_prop(const struct device *dev,
```

Comment:
I like this a lot and it also gives the flexibility of getting multiple block properties if we want if we have a length attribute in the block prop struct. We probably want an example of its usage in that way.